crates/parser/src/grammar/expressions.rs

```diff
@@ -602,14 +603,30 @@ fn arg_list(p: &mut Parser<'_>) {
     // fn main() {
     //     foo(#[attr] 92)
     // }
-    delimited(
-        p,
-        T!['('],
-        T![')'],
-        T![,],
-        EXPR_FIRST.union(ATTRIBUTE_FIRST),
-        |p: &mut Parser<'_>| expr(p).is_some(),
-    );
+    p.bump(T!['(']);
+    while !p.at(T![')']) && !p.at(EOF) {
+        if p.at(T![,]) {
+            // Recover if an argument is missing and only got a delimiter,
+            // e.g. `(a, , b)`.
+            p.error("expected expression");
+            p.bump(T![,]);
+            continue;
+        }
```

Comment:
Can we instead move this into `delimited` as a general mechanism for this? (and then give the function an extra arg for the error message)

Reply:
It makes sense. I'll give it a try.